http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/WsmDao.scala

```diff
@@ -0,0 +1,356 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import java.util.UUID
+
+import ca.mrvisser.sealerate
+import cats.mtl.Ask
+import org.broadinstitute.dsde.workbench.leonardo.{
+  AppContext,
+  AzureCloudContext,
+  CidrIP,
+  DiskSize,
+  ManagedResourceGroupName,
+  RuntimeImage,
+  RuntimeName,
+  SubscriptionId,
+  TenantId,
+  WorkspaceId
+}
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes
+import _root_.io.circe._
+import org.broadinstitute.dsde.workbench.model.{TraceId, WorkbenchEmail}
+
+trait WsmDao[F[_]] {
+  def createIp(request: CreateIpRequest)(implicit ev: Ask[F, AppContext]): F[CreateIpResponse]
+
+  def createDisk(request: CreateDiskRequest)(implicit ev: Ask[F, AppContext]): F[CreateDiskResponse]
+
+  def createNetwork(request: CreateNetworkRequest)(implicit ev: Ask[F, AppContext]): F[CreateNetworkResponse]
+
+  def createVm(request: CreateVmRequest)(implicit ev: Ask[F, AppContext]): F[CreateVmResult]
+
+  def getCreateVmJobResult(request: GetJobResultRequest)(implicit ev: Ask[F, AppContext]): F[CreateVmResult]
+
+  def deleteVm(request: DeleteVmRequest)(implicit ev: Ask[F, AppContext]): F[DeleteVmResult]
+
+  def getWorkspace(workspaceId: WorkspaceId)(implicit ev: Ask[F, AppContext]): F[WorkspaceDescription]
+}
+
+final case class WorkspaceDescription(id: WorkspaceId, displayName: String, azureContext: AzureCloudContext)
+
+//Azure Vm Models
+final case class CreateVmRequest(workspaceId: WorkspaceId,
+                                 common: ControlledResourceCommonFields,
+                                 vmData: CreateVmRequestData)
+
+final case class CreateVmRequestData(name: RuntimeName,
+                                     region: com.azure.core.management.Region,
+                                     vmSize: VirtualMachineSizeTypes,
+                                     vmImageUri: RuntimeImage,
+                                     ipId: WsmControlledResourceId,
+                                     diskId: WsmControlledResourceId,
+                                     networkId: WsmControlledResourceId)
+
+final case class AzureVmName(value: String) extends AnyVal
+final case class AzureImageUri(value: String) extends AnyVal
+
+final case class WsmVm(resourceId: WsmControlledResourceId)
```

Comment:
any suggestions? I feel likes the appropriate ones will surface once I start to work on the unit test code.

Reply:
I would put all value classes in [core](https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala)

This will make later adding automation tests a bit easier